libs/app-shell/src/lib/components/plugin-launcher/plugin-launcher.component.ts

```diff
@@ -101,61 +104,136 @@ export class PluginLauncherComponent implements OnChanges, AfterViewChecked {
             return;
         }
 
-        const _pluginModule = descriptor.modules.find(module => module.name === this.module);
-        const _remoteModule = await loadRemoteModule<DescriptorsModule>(descriptor, _pluginModule)
-            .catch(err => console.error(err));
-        if (!_remoteModule) {
+        const pluginModule = this.findPluginModule(descriptor);
+        const remoteModule = await loadRemoteModule<DescriptorsModule>(descriptor, pluginModule)
+            .catch((error) => {
+                this.dispatchError(error);
+            });
+
+        if (!remoteModule) {
             return;
         }
-        const _module = _remoteModule[_pluginModule.name];
 
-        if (_pluginModule.type !== 'iframe' && !this.iframeUri) {
+        const moduleOrCustomElementName = remoteModule[pluginModule.name];
+        const isNotIframeType = pluginModule.type !== 'iframe' && !this.iframeUri;
+
+        if (isNotIframeType) {
             this._safeIframeUri = null;
         }
 
-        if (_pluginModule.type === 'iframe') {
-            const _url = descriptor.uri + _pluginModule.html;
-            this._safeIframeUri = this.sanitizer.bypassSecurityTrustResourceUrl(_url);
-            return;
-        }
+        switch (pluginModule.type) {
```

Comment:
please don't use `switch`, switch should be used only in pure functions and always return same type
switch for managing logic branches is very hard to support